### PR TITLE
fix: recursive folder scan

### PR DESCRIPTION
**Pull request by TyShkan**
Closes https://github.com/MeltanoLabs/tap-csv/issues/128

Fixes description in meltano.yml (file → path)

### CODE REVIEW

File: tap_csv/client.py [hunk 1]
```diff
@@ -67,10 +67,11 @@ def get_file_paths(self) -> list:
         file_paths = []
         if os.path.isdir(file_path):
             clean_file_path = os.path.normpath(file_path) + os.sep
-            for filename in os.listdir(clean_file_path):
-                file_path = clean_file_path + filename
-                if self.is_valid_filename(file_path):
-                    file_paths.append(file_path)
+            for dirpath, _, filenames in os.walk(clean_file_path):
```
Comment (pnadolny13) [inline tap_csv/client.py:70]: @TyShkan the sonarcloud scans think this method is too complex now. What if we refactor this for loop out to its own private method `_get_recursive_file_paths` or similar that returns a list like

```
        if os.path.isdir(file_path):
            clean_file_path = os.path.normpath(file_path) + os.sep
            file_paths = self._get_recursive_file_paths(clean_file_path)
```

**Comment by pnadolny13** [comment]
@TyShkan this looks great! I ran it locally and added a test to confirm it worked as expected so I opened a PR to your branch with that test https://github.com/TyShkan/tap-csv/pull/1. If you pull that in then I think we're good to merge this!

File: meltano.yml [hunk 2]
```diff
@@ -22,10 +22,10 @@ plugins:
       add_metadata_columns: false
     settings:
       - name: files
-        description: Array of objects containing keys - `entity`, `file`, `keys`, `encoding` (Optional), `delimiter` (Optional), `doublequote` (Optional), `escapechar` (Optional), `quotechar` (Optional), `skipinitialspace` (Optional), `strict` (Optional)
```
Comment (pnadolny13) [inline meltano.yml:25]: @TyShkan good catch!

**Comment by TyShkan** [comment]
Looks like we are good to go!

**Review by pnadolny13 — APPROVED** [review]
@TyShkan great work, thanks for fixing this!

**Pull request merged**